ttkwidgets/numberentry.py

```diff
@@ -0,0 +1,91 @@
+"""
+Author: rdbende
+License: GNU GPLv3
+Copyright (c) 2021 rdbende
+"""
+
+import tkinter as tk
+from tkinter import ttk
+
+
+class NumberEntry(ttk.Entry):
+    """
+    An entry that takes only numbers or calculations and calculates the result of the calculation
+    
+    :param expressions: Allow the use of expressions (default is True)
+    :type expressions: bool
+    
+    :param roundto: The number of decimals in the result (default is 0)
+    :type roundto: int
+    """
+    
+    
+    def __init__(self, master, **kwargs):
+        
+        
+        self._expr = kwargs.pop("expressions", True)
+        self._round = kwargs.pop("roundto", 0)
+        ttk.Entry.__init__(self, master, **kwargs)
+        self.bind("<Return>", self._eval)
+        self.bind("<FocusOut>", self._eval)
+        self.bind("<KeyRelease>", self._check)
+    
+    
+    def _eval(self, *args):
+        current = self.get()
+        try:
+            if len(current) > 0:
+                if int(self._round) == 0:
+                    result = int(round(eval(current), 0))
+                    self.delete(0, "end")
+                    self.insert(0, result)         
+                else:
+                    result = round(float(eval(current)), self._round)
+                    self.delete(0, "end")
+                    self.insert(0, result)
+        except SyntaxError:
+            self.delete(0, "end")
+            self.insert(0, "SyntaxError")
+            self.select_range(0, "end")
+        except ZeroDivisionError:
+            self.delete(0, "end")
+            self.insert(0, "ZeroDivisionError")
+            self.select_range(0, "end")
+        
+        
+    def _check(self, *args):
+        typed = self.get()
+        if not typed == "SyntaxError" and not typed == "ZeroDivisionError":
+            checked = self._replace(typed)
+            self.delete(0, "end")
+            self.insert(0, checked)
+        
+        
+    def _replace(self, typed) -> str:
+        if self._expr:
+            allowed = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "+", "-", "*", "/", "%", "."]
```

Comment:
I'd like there to be some user option to allow specifying more allowed characters, such as maybe scientific prefixes (`m`, `u`, `n`, `p`, `f`, `a` and `k`, `M`, `G` , `T`, `P`, etc.), or for example the `**` (raise-to-power operator).

Reply:
Ok, I'll create user-specified options, but `**` isn't needed because it's still enabled and the floor division `//` also.